eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt

```diff
@@ -18,9 +19,11 @@ class NatsControllerBeanPostProcessor : BeanPostProcessor {
     private fun <ReqT : GeneratedMessageV3, RepT : GeneratedMessageV3>
             NatsController<ReqT, RepT>.initializeNatsController() {
         connection.createDispatcher { message ->
-            val parsedData = parser.parseFrom(message.data)
-            val response = handler(parsedData)
-            connection.publish(message.replyTo, response.toByteArray())
+            handler(parser.parseFrom(message.data))
+                .subscribeOn(Schedulers.boundedElastic())
```

Comment:
double-check that map and doOnNext are running in your boundedElastic treads

Reply:
added .publishOn(Schedulers.boundedElastic())
So all other operators will exec at boundedElastic()